packages/run-protocol/src/reserve/assetReserve.js

```diff
@@ -51,24 +61,70 @@ const nonalphanumeric = /[^A-Za-z0-9]/g;
  *   marshaller: ERef<Marshaller>,
  *   storageNode: ERef<StorageNode>,
  * }} privateArgs
+ * @param {Baggage} baggage
  */
-const start = async (zcf, privateArgs) => {
+const start = async (zcf, privateArgs, baggage) => {
   /**
    * Used to look up the unique keyword for each brand, including RUN.
    *
    * @type {MapStore<Brand, Keyword>}
    */
-  const keywordForBrand = makeStore('keywords');
+  const keywordForBrand = provideDurableMapStore(baggage, 'keywordForBrand');
   /**
    * Used to look up the brands for keywords, excluding RUN because it's a special case.
    *
    * @type {MapStore<Keyword, Brand>}
    */
-  const brandForKeyword = makeStore('brands');
+  const brandForKeyword = provideDurableMapStore(baggage, 'brandForKeyword');
   /**
    * @type {MapStore<Brand, Brand>}
    */
-  const liquidityBrandForBrand = makeStore('liquidityBrands');
+  const liquidityBrandForBrand = provideDurableMapStore(
+    baggage,
+    'liquidityBrandForBrand',
+  );
+
+  const takeFeeMint = async () => {
+    if (baggage.has('runMint')) {
+      return baggage.get('runMint');
+    }
+
+    const runMintTemp = await zcf.registerFeeMint(
```

Comment:
any hazard that this function sometimes returns sync and sometimes awaits? no, I don't think so in this case.

Reply:
I think I responded previously, but I don't see it here, so I'll re-post.

AIUI, no, there's no hazard. If the conditional is true, the caller will quickly get a resolved Promise for the mint retrieved from the baggage. Otherwise they'll have to wait for the round-trip before getting a resolved promise. If they don't need to resolve the promise, they don't have to wait. If they do, they'll add a turn whichever path we took here.